shop.cpp

```diff
@@ -1,26 +1,77 @@
 #include "shop.h"
-
+#include <chrono>
 
 Shop::Shop(int mMaxChairs)
     :mMaxChairs(mMaxChairs)
-    ,mTheBarber(this)
-    ,mBarberSemaphore(0)
+    ,mBarberSemaphore(1)
     ,mCustomersSemaphore(0)
 {
+    mTheBarber = new Barber(this);
+}
+
+Shop::~Shop()
+{
+    delete mTheBarber;
+    mTheCustomers.clear();
```

Comment:
Who is deleting the customers?

Reply:
Call is invoked by the Shop. You can see main function.